packages/keystone/plugins/utils/Relationship.js

```diff
@@ -0,0 +1,54 @@
+const {
+    Relationship: RelationshipBase,
+} = require('@keystonejs/fields')
+const {
+    AuthedRelationship: AuthedRelationshipBase,
+} = require('@keystonejs/fields-authed-relationship')
+const { get, isEmpty } = require('lodash')
+
+async function internalResolveNestedOperations (
+    resolveNestedOperations, operations, item, context, getItem, mutationState
+) {
+    // we have a special case for skipResolveRelations flag
+    // in case if we are upsert relation field and skipResolveRelations is set up
+    // let's skip resolving nested connect operation
+    const connectId = get(operations, 'connect.id')
+    if (context.skipAccessControl && Object.keys(operations).length === 1 && !isEmpty(connectId)) {
```

Comment:
What's length check is used for?

Reply:
Just to make sure operations hold only connect instruction